### PR TITLE
Adds some information on the MessageTypeToSend

### PR DESCRIPTION
**Pull request by mauroservienti**
@SimonCropp @indualagarsamy I tried to expand a little bit on the `MessageTypeToSend` dictionary value, I'm not super happy with the result, can you please chime in?

### CODE REVIEW

File: nservicebus/sqlserver/operations-scripting.md [hunk 1]
```diff
@@ -31,6 +31,10 @@ snippet:sqlserver-powershell-nativesend
 
 snippet:sqlserver-nativesend-usage
 
+The `MessageTypeToSend` dictionary value represents the message .NET type, by default the endpoint receiving the message will use that message type to deserialize the incoming message. Depending on the [versioning](/samples/versioning/sample.md) strategy the type can be expressed in 2 different ways:
```
Comment (weralabaj) [inline nservicebus/sqlserver/operations-scripting.md]: > The `MessageTypeToSend` dictionary value represents the message .NET type, by default the endpoint receiving the message will use that message type to deserialize the incoming message. Depending on the [versioning](/samples/versioning/sample.md) strategy the type can be expressed in 2 different ways:

The `MessageTypeToSend` ~~dictionary~~ value is the .NET type of the message. The receiving endpoint will use the type when deserializing incoming message. Depending on the [versioning](/samples/versioning/sample.md) strategy the type can be specified in the following ways:

File: nservicebus/sqlserver/operations-scripting.md [hunk 2]
```diff
@@ -31,6 +31,10 @@ snippet:sqlserver-powershell-nativesend
...
+
+* Full type name in the form of `Namespace.ClassName, AssemblyName`
+* Assembly qualified name in the form of `Namespace.ClassName, AssemblyName, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null`
```
Comment (weralabaj) [inline nservicebus/sqlserver/operations-scripting.md]: >* Full type name in the form of `Namespace.ClassName, AssemblyName`
>* Assembly qualified name in the form of `Namespace.ClassName, AssemblyName, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null`

* Full type name: `Namespace.ClassName, AssemblyName`
* Assembly qualified name: `Namespace.ClassName, AssemblyName, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null`

File: nservicebus/sqlserver/operations-scripting.md [hunk 3]
```diff
@@ -31,6 +31,10 @@ snippet:sqlserver-powershell-nativesend
 
 snippet:sqlserver-nativesend-usage
 
+The `MessageTypeToSend` dictionary value represents the message .NET type, by default the endpoint receiving the message will use that message type to deserialize the incoming message. Depending on the [versioning](/samples/versioning/sample.md) strategy the type can be expressed in 2 different ways:
+
+* Full type name in the form of `Namespace.ClassName, AssemblyName`
```
Comment (andreasohlund) [inline nservicebus/sqlserver/operations-scripting.md]: Full type name should not include the asm, `Namespace.ClassName` is enough

Comment (andreasohlund) [inline nservicebus/sqlserver/operations-scripting.md]: Also should we mention that you can leave out the header completely if 

1. You are using XML and the root node == messagetype
2. json.net and $type=messagetype
?

File: nservicebus/sqlserver/operations-scripting.md (hunk 1 above)
Comment (SimonCropp) [inline nservicebus/sqlserver/operations-scripting.md]: i actually dont think this content should be in operations-scripting.md. perhaps somewhere in the serialization doco or the header doco. then link to that from here. @mauroservienti thoughts?

Comment (indualagarsamy) [inline nservicebus/sqlserver/operations-scripting.md]: I agree with @SimonCropp. 
@mauroservienti - I think we can try and add that info into this doco: http://docs.particular.net/nservicebus/messaging/message-owner

Thoughts?

Comment (weralabaj) [inline nservicebus/sqlserver/operations-scripting.md]: @indualagarsamy I think here would make even more sense http://docs.particular.net/nservicebus/messaging/headers. The message owner doco needs update too (snippets?), but this particular piece of info would fit headers better imo.

Comment (indualagarsamy) [inline nservicebus/sqlserver/operations-scripting.md]: @weralabaj - sounds good.

**Comment by indualagarsamy** [comment]
@mauroservienti - any updates on this?

File: nservicebus/messaging/headers.md [hunk 4]
```diff
@@ -34,30 +34,50 @@ public static DateTime ToUtcDateTime(string wireFormattedString)
...
+### NServiceBus.ContentType
+The type of serialization used for the message. For example ` text/xml` or `text/json`. The `NServiceBus.ContentType` header was added in Version 4.0. In some cases it may be useful to use the `NServiceBus.Version` header to determine when to use the `NServiceBus.ContentType` header.
+
+### NServiceBus.EnclosedMessageTypes
+The fully qualified .NET type name of the enclosed message(s). The receiving endpoint will use this type when deserializing an incoming message. Depending on the [versioning strategy](/samples/versioning/) the type can be specified in the following ways:
+
+* Full type name: `Namespace.ClassName`
+* Assembly qualified name: `Namespace.ClassName, AssemblyName, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null`
 
+NOTE: In integration scenarios, this header can be safely omitted if the endpoint uses XmlSerialization and the root node happens to message type.
```
Comment (indualagarsamy) [inline nservicebus/messaging/headers.md]: I was trying to incorporate @andreasohlund's comment here, however I am not sure how to make it sound right for the Json serialization. @Particular/docs-maintainers - feel free to fix up the note. 

>Also should we mention that you can leave out the header completely if
You are using XML and the root node == messagetype
json.net and $type=messagetype ?

**Comment by indualagarsamy** [comment]
@Particular/docs-maintainers - can you review this?

Also need to address the NOTE in headers.md - for the `EnclosedMessageTypes` header on the Json serialization with what @andreasohlund mentioned earlier 

>Also should we mention that you can leave out the header completely if
You are using XML and the root node == messagetype
json.net and $type=messagetype ?

File: nservicebus/messaging/headers.md [hunk 5]
```diff
@@ -34,30 +34,50 @@ public static DateTime ToUtcDateTime(string wireFormattedString)
...
+### NServiceBus.ContentType
+The type of serialization used for the message. For example ` text/xml` or `text/json`. The `NServiceBus.ContentType` header was added in Version 4.0. In some cases it may be useful to use the `NServiceBus.Version` header to determine when to use the `NServiceBus.ContentType` header.
+
+### NServiceBus.EnclosedMessageTypes
+The fully qualified .NET type name of the enclosed message(s). The receiving endpoint will use this type when deserializing an incoming message. Depending on the [versioning strategy](/samples/versioning/) the type can be specified in the following ways:
+
+* Full type name: `Namespace.ClassName`
+* Assembly qualified name: `Namespace.ClassName, AssemblyName, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null`
 
+NOTE: In integration scenarios, this header can be safely omitted if the endpoint uses XmlSerialization and the root node happens to be the message type.
```
Comment (SimonCropp) [inline nservicebus/messaging/headers.md]: not sure i like "happens to be"? wont this always be the case for the xml serializer?

File: nservicebus/messaging/headers.md [hunk 6]
```diff
@@ -34,30 +34,50 @@ public static DateTime ToUtcDateTime(string wireFormattedString)
...
- * `NServiceBus.RelatedTo`: The `MessageId` that caused this message to be sent
- * `NServiceBus.MessageIntent`: Can be one of the following:
-	* `Send`: Regular point-to-point send. Note that messages sent to Error queue will also have a `Send` intent.
-	* `Publish`: The message is an event that has been published and will be sent to all subscribers.
-	* `Subscribe`: A control message indicating that the source endpoint would like to subscribe to a specific message.
-	* `Unsubscribe`: A control message indicating that the source endpoint would like to unsubscribe to a specific message.
-	* `Reply`: The message has been initiated by doing a Reply or a Return from within a Handler or a Saga.
- * `NServiceBus.ReplyToAddress`: The queue address that instructs downstream handlers or sagas where to send to when doing a Reply or Return.
+### NServiceBus.MessageId
+A unique id for the current message. Note that the value used for an outgoing message can be controlled by the endpoint, using an `IMutateOutgoingTransportMessages`.
```
Comment (SimonCropp) [inline nservicebus/messaging/headers.md]: link to mutators instead of using the type?

File: nservicebus/messaging/headers.md [hunk 7]
```diff
@@ -34,30 +34,50 @@ public static DateTime ToUtcDateTime(string wireFormattedString)
...
+
+### NServiceBus.CorrelationId
+A string used to [correlate](./message-correlation.md) reply messages to their corresponding request message.
```
Comment (SimonCropp) [inline nservicebus/messaging/headers.md]: `./` is redundant

File: nservicebus/messaging/headers.md [hunk 8]
```diff
@@ -34,30 +34,50 @@ public static DateTime ToUtcDateTime(string wireFormattedString)
...
+
+### NServiceBus.ConversationId
+The conversation that this message is part of
```
Comment (SimonCropp) [inline nservicebus/messaging/headers.md]: full stop

File: nservicebus/messaging/headers.md [hunk 9]
```diff
@@ -34,30 +34,50 @@ public static DateTime ToUtcDateTime(string wireFormattedString)
...
+
+### NServiceBus.RelatedTo
+The `MessageId` that caused this message to be sent
```
Comment (SimonCropp) [inline nservicebus/messaging/headers.md]: full stop

File: nservicebus/messaging/headers.md (hunk 5 above)
Comment (indualagarsamy) [inline nservicebus/messaging/headers.md]: @andreasohlund - please suggest what should go in here and also about the Json serializer

Comment (indualagarsamy) [inline nservicebus/messaging/headers.md]: @simoncropp - not always. I think it depends on the NSB Version.

File: nservicebus/messaging/headers.md (hunk 6 above)
Comment (indualagarsamy) [inline nservicebus/messaging/headers.md]: Good idea. Will do.

**Comment by SimonCropp** [comment]
merged manually here https://github.com/Particular/docs.particular.net/commit/1faa607291216a937030b2ceaebfcef17f08e4cc

@mauroservienti @indualagarsamy thanks for your help on this one

**Pull request closed without merging**